### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by andrewjl-mux**
- Update Mux Data AVPlayer and Core SDK dependencies
- Replace Specta unit tests with XCTest
- Clean-up

### CODE REVIEW

File: Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m [hunk 1]
```diff
@@ -64,58 +64,100 @@ - (void)setupAdViewData:(MUXSDKAdEvent *)event withAd:(IMAAd *)ad {
...
+            playbackEvent = [[MUXSDKAdPauseEvent alloc] init];
             break;
-        case kIMAAdEvent_RESUME:
-            playbackEvent = [MUXSDKAdPlayEvent new];
-            [self setupAdViewData:playbackEvent withAd:event.ad];
+        }
+        case kIMAAdEvent_RESUME: {
+            playbackEvent = [[MUXSDKAdPlayEvent alloc] init];
+            MUXSDKViewData *viewData = [[MUXSDKViewData alloc] init];
+            if ([_playerBinding getCurrentPlayheadTimeMs] < 1000) {
```
Comment (cjpillsbury) [inline Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m:128]: question: I remember customers asking for this value to be configurable. Does anyone recall where we landed on that? (@andrewjl-mux @daytime-em)

File: Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m [hunk 2]
```diff
@@ -64,58 +64,100 @@ - (void)setupAdViewData:(MUXSDKAdEvent *)event withAd:(IMAAd *)ad {
...
             }
             break;
+        }
         default:
             break;
     }
+
     if (playbackEvent != nil) {
-        [self setupAdViewData:playbackEvent withAd:event.ad];
+        MUXSDKViewData *viewData = [[MUXSDKViewData alloc] init];
```
[the remark this answers is not in the document]
Comment (andrewjl-mux) [reply Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m:154]: This shouldn't change anything.

File: Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m [hunk 3]
```diff
@@ -64,58 +64,100 @@ - (void)setupAdViewData:(MUXSDKAdEvent *)event withAd:(IMAAd *)ad {
...
+        case kIMAAdEvent_STARTED: {
             if (_sendAdplayOnStarted) {
-                playbackEvent = [MUXSDKAdPlayEvent new];
+                playbackEvent = [[MUXSDKAdPlayEvent alloc] init];
                 [_playerBinding dispatchAdEvent: playbackEvent];
             } else {
                 _sendAdplayOnStarted = YES;
             }
-            playbackEvent = [MUXSDKAdPlayingEvent new];
+            playbackEvent = [[MUXSDKAdPlayingEvent alloc] init];
```
Comment (andrewjl-mux) [inline Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m:98]: Calling `alloc` and `init` is equivalent to `new`. `new` doesn't support custom initializers (initializers other than `init`, typically these take an argument). Modern Objective-C recommends use of custom initializers or factory methods (`+ (instancetype)factoryMethodA`) for more explicitness. [See here.](https://developer.apple.com/library/archive/releasenotes/ObjectiveC/ModernizationObjC/AdoptingModernObjective-C/AdoptingModernObjective-C.html#//apple_ref/doc/uid/TP40014150-CH1-SW8).

File: Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m (hunk 1 above)
Comment (andrewjl-mux) [reply Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m:128]: Makes sense to do. Better as a separate change.

File: Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m (hunk 3 above)
Comment (andrewjl-mux) [reply Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m:98]: Everything here [is compiled with Automatic Reference Counting enabled](https://clang.llvm.org/docs/AutomaticReferenceCounting.html), which disallows explicit management of reference counts except in limited circumstances. Bridging calls to older APIs. Not used in this SDK.

**Comment by andrewjl-mux** [comment]
Release branch PR: https://github.com/muxinc/mux-stats-google-ima/pull/40

**Comment by andrewjl-mux** [comment]
> LGTM!
> 
> Had a non-blocking question about whether we should remove `Podfile.lock`

https://github.com/muxinc/mux-stats-google-ima/pull/38#discussion_r1783194194

File: Example/Podfile.lock
[the remark this answers is not in the document]
Comment (daytime-em) [reply Example/Podfile.lock:1]: Ahh gotcha. Not as simple as I thought :)

Thanks for the explanation!

**Pull request merged**